### PR TITLE
feat: provide proper network timings

### PR DESCRIPTION
**Pull request by sadym-chromium**
+ make e2e test offline.

Tested manually + WPT.

### CODE REVIEW

File: src/bidiMapper/modules/network/NetworkRequest.ts [hunk 1]
```diff
@@ -304,26 +304,66 @@ export class NetworkRequest {
...
       // TODO: Verify this is correct
       // https://source.chromium.org/chromium/chromium/src/+/main:net/base/load_timing_info.h;l=145
-      fetchStart: getTiming(this.#response.info?.timing?.requestTime),
-      dnsStart: getTiming(this.#response.info?.timing?.dnsStart),
-      dnsEnd: getTiming(this.#response.info?.timing?.dnsEnd),
-      connectStart: getTiming(this.#response.info?.timing?.connectStart),
-      connectEnd: getTiming(this.#response.info?.timing?.connectEnd),
-      tlsStart: getTiming(this.#response.info?.timing?.sslStart),
-      requestStart: getTiming(this.#response.info?.timing?.sendStart),
+      fetchStart: getTiming(
```
Comment (OrKoN) [inline src/bidiMapper/modules/network/NetworkRequest.ts:329]: I think these have to be durations from timeOrigin

File: src/bidiMapper/modules/network/NetworkRequest.ts [hunk 2]
```diff
@@ -304,26 +304,66 @@ export class NetworkRequest {
...
+    const responseTimeOffset = getTiming(
+      getTiming(this.#response.info?.timing?.requestTime) -
+        getTiming(this.#request.info?.timestamp),
+    );
+
     return {
       // TODO: Verify this is correct
-      timeOrigin: getTiming(this.#response.info?.timing?.requestTime),
-      requestTime: getTiming(this.#response.info?.timing?.requestTime),
+      timeOrigin: Math.round(getTiming(this.#request.info?.wallTime) * 1000),
```
Comment (OrKoN) [inline src/bidiMapper/modules/network/NetworkRequest.ts:319]: I think timeOrigin is the time of the creation of the navigable. I think we should take the navigation request wallTime for that.

**Comment by OrKoN** [comment]
I think failures are due to the ubuntu 24.04 rollout (we probably need this https://github.com/puppeteer/puppeteer/blob/main/.github/workflows/ci.yml#L138)

**Review by OrKoN — APPROVED** [review]

File: src/bidiMapper/modules/network/NetworkRequest.ts (hunk 1 above)
Comment (sadym-chromium) [reply src/bidiMapper/modules/network/NetworkRequest.ts:329]: I assume CDP `response.timing` is relative to the baseline CDP `response.timing.requestTime`. The `responseTimeOffset` represents the diff between request and response baseline. It's my speculations which ends up to a meaningful result.

File: src/bidiMapper/modules/network/NetworkRequest.ts (hunk 2 above)
Comment (sadym-chromium) [reply src/bidiMapper/modules/network/NetworkRequest.ts:319]: I'm not sure if we can reliably get the time of the creation of the navigable. Especially if it was created before the Mapper.

Comment (OrKoN) [reply src/bidiMapper/modules/network/NetworkRequest.ts:319]: it is performance.timeOrigin in the web APIs

**Pull request merged**